vm/src/stdlib/time.rs

```diff
@@ -79,409 +29,515 @@ pub fn get_time() -> f64 {
+            (intpart * mul + remaining) as u64
         }
 
-        (time_from_filtime(kernel_time), time_from_filtime(user_time))
-    };
-    Ok(Duration::from_nanos((ktime + utime) * 100))
-}
-
-#[cfg(not(windows))]
-fn get_process_time(vm: &VirtualMachine) -> PyResult<Duration> {
-    cfg_if::cfg_if! {
-        if #[cfg(any(
-            target_os = "android",
-            target_os = "dragonfly",
-            target_os = "freebsd",
-            target_os = "linux"
-        ))] {
-            let mut time = libc::timespec {
-                tv_sec: 0,
-                tv_nsec: 0,
-            };
-
-            if unsafe { libc::clock_gettime(libc::CLOCK_PROCESS_CPUTIME_ID, &mut time) } == -1 {
+        let t: libc::tms = unsafe {
+            let mut t = std::mem::MaybeUninit::uninit();
+            if libc::times(t.as_mut_ptr()) == -1 {
                 return Err(vm.new_os_error("Failed to get clock time".to_owned()));
             }
+            t.assume_init()
+        };
 
-            Ok(Duration::new(time.tv_sec as u64, time.tv_nsec as u32))
-        } else if #[cfg(any(
-            target_os = "illumos",
-            target_os = "netbsd",
-            target_os = "solaris",
-            target_os = "openbsd"
-        ))] {
-            let mut ru: libc::rusage = unsafe { std::mem::zeroed() };
-            if unsafe { libc::getrusage(libc::RUSAGE_SELF, &mut ru) } == -1 {
-                return Err(vm.new_os_error("Failed to get clock time".to_owned()));
-            }
+        #[cfg(target_os = "wasi")]
+        let freq = 60;
+        #[cfg(not(target_os = "wasi"))]
+        let freq = unsafe { libc::sysconf(libc::_SC_CLK_TCK) };
 
-            let utime = time_fromtimeval(ru.ru_utime, vm)?;
-            let stime = time_fromtimeval(ru.ru_stime, vm)?;
+        Ok(std::time::Duration::from_nanos(
+            time_muldiv(t.tms_utime, SEC_TO_NS, freq) + time_muldiv(t.tms_stime, SEC_TO_NS, freq),
+        ))
+    }
 
-            Ok(Duration::from_nanos(utime + stime))
-        } else if #[cfg(any(
-            all(target_arch = "wasm32", not(target_os = "unknown")),
-            target_os = "redox"
-        ))] {
-            let mut t: libc::tms = unsafe { std::mem::zeroed() };
-            if unsafe { libc::times(&mut t) } == -1 {
-                return Err(vm.new_os_error("Failed to get clock time".to_owned()));
-            }
+    #[cfg(not(any(
+        windows,
+        target_os = "macos",
+        target_os = "android",
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "linux",
+        target_os = "illumos",
+        target_os = "netbsd",
+        target_os = "solaris",
+        target_os = "openbsd",
+        target_os = "redox",
+        all(target_arch = "wasm32", not(target_os = "unknown"))
+    )))]
+    fn get_process_time(vm: &VirtualMachine) -> PyResult<std::time::Duration> {
+        Err(vm.new_not_implemented_error("thread time unsupported in this system".to_owned()))
```

Comment:
```suggestion
        Err(vm.new_not_implemented_error("process time unsupported in this system".to_owned()))
```
Maybe this is typo :)